homework-05/src/main.rs

```diff
@@ -0,0 +1,172 @@
+use slug::slugify;
+use std::env;
+use std::collections::HashMap;
+use std::error::Error;
+use std::io;
+use std::io::Write;
+use csv::Reader;
+
+fn lowercase(input: &str) -> Result<String, Box<dyn Error>> {
+    Ok(input.to_lowercase())
+}
+
+fn alphabetic_lowercase(input: &str) -> Result<String, Box<dyn Error>> {
+    if input.chars().any(|c| !c.is_alphabetic()) {
+        return Err("Input contains non-alphabetic characters".into());
+    }
+    Ok(input.to_lowercase())
+}
+
+fn uppercase(input: &str) -> Result<String, Box<dyn Error>> {
+    Ok(input.to_uppercase())
+}
+
+fn alphabetic_uppercase(input: &str) -> Result<String, Box<dyn Error>> {
+    if input.chars().any(|c| !c.is_alphabetic()) {
+        return Err("Input contains non-alphabetic characters".into());
+    }
+    Ok(input.to_lowercase())
+}
+
+fn no_spaces(input: &str) -> Result<String, Box<dyn Error>> {
+    Ok(input.replace(" ", ""))
```

Comment:
Whenever you can use a char as pattern as opposed to str, it is marginally faster to do that. If I recall, there is a clippy lint for it! (probably enabled by default)